src/pure/ketrew_host.mli

```diff
@@ -0,0 +1,133 @@
+(**************************************************************************)
+(*  Copyright 2014, Sebastien Mondet <seb@mondet.org>                     *)
+(*                                                                        *)
+(*  Licensed under the Apache License, Version 2.0 (the "License");       *)
+(*  you may not use this file except in compliance with the License.      *)
+(*  You may obtain a copy of the License at                               *)
+(*                                                                        *)
+(*      http://www.apache.org/licenses/LICENSE-2.0                        *)
+(*                                                                        *)
+(*  Unless required by applicable law or agreed to in writing, software   *)
+(*  distributed under the License is distributed on an "AS IS" BASIS,     *)
+(*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or       *)
+(*  implied.  See the License for the specific language governing         *)
+(*  permissions and limitations under the License.                        *)
+(**************************************************************************)
+
+(** Definition of a host; a place to run commands or handle files. *)
+open Ketrew_pervasives
+
+(** Definitions specific to “SSH” hosts (see {!connection}). *)
+module Ssh : sig
+
+  type t = {
+    address: string;
+    port: int option;
+    user: string option;
+    add_ssh_options: string list;
+  } [@@deriving yojson]
+  (** The type of SSH-based hosts. *)
+
+  val configure_ssh_batch_option :
+    [ `Custom of string | `Dropbear | `Openssh ] -> unit
+    (** Configure global “Batch option”,
+      (call [ssh] without password/question):
+      {ul
+        {li for OpenSSH, it is ["-oBatchMode=yes"],}
+        {li for DropBear, it is ["-s"].  }
+      }*)
+
+  val ssh_batch_option: t -> string
+  (** Get the right option for the SSH client, for now this does not
+      actually depend on the Host. *)
```

Comment:
Yes, this was already there before the PR; github highlighted the diff because of the `git mv`.

For `ssh_batch_option` is a global variable, I wanted to add it to the `Ssh.t` later (trying to make this PR change the types as little as possible).